common/src/main/java/org/apache/falcon/entity/parser/ClusterEntityParser.java

```diff
@@ -243,58 +244,95 @@ private void validateLocations(Cluster cluster) throws ValidationException {
         try {
             fs = HadoopClientFactory.get().createProxiedFileSystem(conf);
         } catch (FalconException e) {
-            throw new ValidationException(
-                    "Unable to get file system handle for cluster " + cluster.getName(), e);
+            throw new ValidationException("Unable to get file system handle for cluster " + cluster.getName(), e);
         }
 
-        for (Location location : cluster.getLocations().getLocations()) {
-            final String locationName = location.getName();
-            if (locationName.equals("temp")) {
-                continue;
-            }
+        Location stagingLocation = ClusterHelper.getLocation(cluster, ClusterLocationType.STAGING);
 
-            try {
-                checkPathOwnerAndPermission(cluster.getName(), location.getPath(), fs,
-                        "staging".equals(locationName)
-                                ? HadoopClientFactory.ALL_PERMISSION
-                                : HadoopClientFactory.READ_EXECUTE_PERMISSION);
-            } catch (IOException e) {
-                throw new ValidationException("Unable to validate the location of name: "
-                        + location.getName() + " with path:" + location.getPath()
-                        + " for cluster " + cluster.getName(), e);
+        if (stagingLocation == null) {
```

Comment:
Yup I have changed the perms check from 777 (All perms) to atleast 755 for staging perms as per
https://issues.apache.org/jira/browse/FALCON-910?focusedCommentId=14227619&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14227619 

Regarding perms this what i am doing:
https://issues.apache.org/jira/browse/FALCON-910?focusedCommentId=14255793&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14255793